src/test/java/org/apache/commons/io/FileUtilsListFilesTest.java

```diff
@@ -222,4 +219,48 @@ public void testListFilesWithDeletion() throws IOException {
         assertEquals(4, list.size());
     }
 
+    @Test
+    public void testListFilesWithDeletionThreaded() throws java.util.concurrent.ExecutionException, InterruptedException {
+        // test for IO-856
+        // create random directory in tmp, create the directory if it does not exist
+        final File dir = FileUtils.getTempDirectory();
+        if (!dir.exists()) {
+            if (!dir.mkdirs()) {
+                throw new RuntimeException("could not create image file path: " + dir.getAbsolutePath());
+            }
+        }
+        final int waitTime = 10000;
+        final java.util.concurrent.CompletableFuture<Void> c1 = java.util.concurrent.CompletableFuture.runAsync(() -> {
+            long endTime = System.currentTimeMillis() + waitTime;
+            while (System.currentTimeMillis() < endTime) {
+                final File file = new File(dir.getAbsolutePath(), java.util.UUID.randomUUID() + ".deletetester");
+                file.deleteOnExit();
+                try {
+                    new java.io.BufferedOutputStream(java.nio.file.Files.newOutputStream(file.toPath())).write("TEST".getBytes(java.nio.charset.StandardCharsets.UTF_8));
+                } catch (Exception e) {
+                    throw new RuntimeException("could not create test file: " + file.getAbsolutePath(), e);
+                }
+                if (!file.delete()) {
+                    throw new RuntimeException("could not delete test file: " + file.getAbsolutePath());
+                }
+            }
+        });
+
+        final java.util.concurrent.CompletableFuture<Void> c2 = java.util.concurrent.CompletableFuture.runAsync(() -> {
+            long endTime = System.currentTimeMillis() + waitTime;
+            try {
+                while (System.currentTimeMillis() < endTime) {
+                    final Collection<File> files = FileUtils.listFiles(dir, new String[]{"\\.deletetester"}, false);
+                    files.size();
```

Comment:
What does calling `size()` do here? Should there be an assert for the result? is the call needed at all?